src/lib/operators/insert.cpp

```diff
@@ -128,7 +128,18 @@ std::shared_ptr<const Table> Insert::_on_execute(std::shared_ptr<TransactionCont
       // Grow MVCC vectors and mark new (but still empty) rows as being under modification by current transaction.
       // Do so before resizing the Segments, because the resize of `Chunk::_segments.front()` is what releases the
       // new row count.
-      target_chunk->get_scoped_mvcc_data_lock()->grow_by(num_rows_for_target_chunk, context->transaction_id());
+      {
```

Comment:
This should be done outside of the whole-table lock then, right?

Reply:
Can't. If we shrink the vector, `mvcc_data->tids` might become invalid.

I'm thinking more and more that this whole locking thing is quite dangerous (see #1602).